content/authoring/kata/adding-images.md

```diff
@@ -0,0 +1,81 @@
+---
+kind: recipe
+---
+
+# Adding Images to Kata Descriptions
+
+Currently, it's not possible to attach any files or additional assets to a kata, but many authors would like to illustrate their descriptions with images. There are plans to introduce the possibility of uploading and attaching files to descriptions, but until it's implemented, authors have to resort to other techniques. There are a few of them, and each of them has its advantages and disadvantages.
+
+:::note
+Images used to be necessary to present equations and formulas. Since support for KaTeX was introduced into Codewars, it's no longer the case. Math formulas can be embedded into descriptions using [Codewars Support for Math Typesetting][markdown-extension-math].
+:::
+
+
+## Avoid Images
+
+While eye-catching and potentially helpful, images used in kata descriptions often cause problems:
+
+- Some authors mistakenly think that illustrative images may be used in place of a clear and concise description. However, images may become unavailable over time for various reasons, so they should never contain information critical to understanding the kata not mentioned elsewhere in the description.
+- Images incur additional maintenance costs. Descriptions can be easily updated, but images have to be replaced on each modification. When a description is modified but an image is not, they become out of sync.
+- Images cause problems with layout. It's difficult to get their size right, they look differently for users with different screen sizes, resolutions, or screen orientations. An image can appear to look nice on a computer screen, but can break things for mobile users.
+- Authors often miss the fact that images can look bad when combined with theming. When an image looks good when viewed with a dark theme, it can look really bad in light mode, and vice versa.
+- Images are subject to licensing, copyright, and intellectual property rights. Authors happen to use images they are not allowed to, or fail to give proper credit.
+
+However, if you are sure that adding an image to a kata description is a good idea, there are some options available.
+
+
+## ASCII-art
+
+ASCII-art is able to represent surprisingly many concepts, being quite simple at the same time. It's easy to add to the description, all it takes is to surround a block of characters with code markdown:
+
+~~~
+```
++----------+    +-------+             +-------+
+|          |    | inner |             |       |
+|   head   +--->+ node  +---> ... +-->+ tail  |
+|          |    |       |             |       |
++----------+    +-------+             +-------+
+```
+~~~
+
+Drawing ASCII images can be supported by a variety of tools (for example, [ASCIIflow](http://asciiflow.com/)). It can be sufficient for representing simple charts and diagrams, but won't be enough for more detailed or colorfum images.
```

Comment:
```suggestion
Drawing ASCII images can be supported by a variety of tools (for example, [ASCIIflow](http://asciiflow.com/)). It can be sufficient for representing simple charts and diagrams, but won't be enough for more detailed or colorful images.
```